v2/router_middleware.go

```diff
@@ -53,7 +53,7 @@ func (r routeTableWithMiddleware) Procedures() []Procedure {
 	return r.m.Procedures(r.r)
 }
 
-func (r routeTableWithMiddleware) Choose(ctx context.Context, req *Request) (HandlerSpec, error) {
```

Comment:
Do we need to rename Router to TransportRouter, to distinguish EncodingRouter?

Reply:
I'm doing that in the following diff, where EncodingRouter is introduced